application/src/main/java/org/togetherjava/tjbot/features/reminder/RemindRoutine.java

```diff
@@ -142,4 +136,32 @@ static ReminderRoute toPrivate(PrivateChannel channel) {
                             + " the original channel you wanted it to be send to)");
         }
     }
+
+    private void attemptRetryReminder(Reminder reminder, Throwable failure) {
+        if (reminder.pendingReminders().getFailureAttempts() > MAX_FAILURE_RETRY) {
+            logger
+                .warn("""
+                        Failed to send a reminder with (authorID '{}') skipping it. This can be due to a network issue, \
+                        but also happen if the bot disconnected from the target guild and the \
+                        user has disabled DMs or has been deleted. It's (content includes '{}' with reminderId '{}'). The reason for (failure '{}')""",
+                        reminder.pendingReminders().getAuthorId(),
+                        reminder.pendingReminders().getContent(),
+                        reminder.pendingReminders().getId(), failure);
+            return;
+
+        }
+
+        int failureAttempts = reminder.pendingReminders().getFailureAttempts() + 1;
+        Instant remindAt = Instant.now().plusSeconds(60);
+        database.write(context -> context.newRecord(PENDING_REMINDERS)
+            .setId(reminder.pendingReminders().getId())
+            .setCreatedAt(Instant.from(reminder.pendingReminders().getCreatedAt()))
+            .setGuildId(reminder.pendingReminders().getGuildId())
+            .setChannelId(reminder.pendingReminders().getChannelId())
+            .setAuthorId(reminder.pendingReminders().getAuthorId())
+            .setRemindAt(remindAt)
+            .setContent(reminder.pendingReminders().getContent())
+            .setFailureAttempts(failureAttempts)
+            .insert());
```

Comment:
y u need to add it to the db?? just attempt 3 times in ur code

Reply:
@Taz03 we decided on that early on. its the easiest approach and also works if the bot might shutdown or crash in between.

if u retry in-code, u have to add the delay (+1 minute) manually and create a proper retry-loop. thats not trivial at all. its much easier to just let the system pick the reminder up again naturally by simply reinserting 🤷

Reply:
that's what issue suggested, adding a separate column to keep track of `failure_attempts`